### PR TITLE
Only Normalize Unsigned Int Types in Prediction

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

Normalization is only performed for unsigned int types in the prediction stage.

### Checklist

- [x] Updated `docs/changelog.rst`
- [x] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

### CODE REVIEW

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py [hunk 1]
```diff
@@ -715,7 +715,8 @@ def numpy_predict(self, x: np.ndarray, raw_out: bool = False) -> Any:
         x = torch.tensor(x)
         x = self.to_batch(x)
         x = x.permute((0, 3, 1, 2))
-        out = self.predict(x, normalize=True, raw_out=raw_out)
+        out = self.predict(
```
Comment (jamesmcclain) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py:718]: It might make sense to cast `x` to type float (float32) in `predict` when normalization is not done because it seems that `x` must be the same type as the weights.

Comment (lewfish) [reply rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py:718]: Sounds good

**Comment by lewfish** [comment]
Why / when would a model not be trained with normalization? Is it just if the model was trained outside of RV? The semantic segmentation DataSet is doing normalization based on the dtype of the imagery here https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py#L82-L84 It seem like we should just normalize the same way at prediction time as at training time, but that doesn't seem to be the case. Am I missing something @AdeelH? I'm not opposed to adding this option, but it seems like a better solution in most cases is to do normalization the same way for prediction.

**Comment by AdeelH** [comment]
> It seem like we should just normalize the same way at prediction time as at training time

I agree. It makes sense to do that in `Learner.numpy_predict()`. I think `Learner.predict()` should not be responsible for normalization at all and should expect its input to be in the right format already.

**Comment by jamesmcclain** [comment]
> Why / when would a model not be trained with normalization? Is it just if the model was trained outside of RV? The semantic segmentation DataSet is doing normalization based on the dtype of the imagery here https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py#L82-L84 It seem like we should just normalize the same way at prediction time as at training time, but that doesn't seem to be the case. Am I missing something @AdeelH? I'm not opposed to adding this option, but it seems like a better solution in most cases is to do normalization the same way for prediction.

The motivation for this is that I noticed normalization was not being done when I was working on `float32` rasters.  I am open to any solution that allows the `predict` stage to give me correct answers.

**Comment by jamesmcclain** [comment]
@lewfish This PR has been updated in accordance with our conversation.

**Review by lewfish — APPROVED** [review]

**Review by lewfish — CHANGES_REQUESTED** [review]
Oops, I'm cancelling that approval. For once I think the integration tests are telling us something useful. I'm guessing that the input was already converted to floats before passing it into the normalize function which is causing it to bypass the normalization which is necessary.

**Comment by lewfish** [comment]
I'm looking into how to fix this now...

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py [hunk 2]
```diff
@@ -664,7 +664,11 @@ def normalize_input(self, x: Tensor) -> Tensor:
             the same tensor that has been scaled to [0-1].
 
         """
-        return x.float() / 255.0
+        if x.dtype == np.uint8 or x.dtype == np.uint16 or x.dtype == np.uint32:
```
Comment (lewfish) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py:667]: x is a torch tensor so you need to use torch.uint8, etc or the equality check won't work.

**Review by lewfish — CHANGES_REQUESTED** [review]

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py (hunk 2 above)
Comment (jamesmcclain) [reply rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py]: Hmmm ... it seems as though `torch.uint16` and `torch.uint32` do not exist.

**Comment by AdeelH** [comment]
@jamesmcclain I made a PR to this branch https://github.com/jamesmcclain/raster-vision/pull/2 uses the same type conversion code as `SemanticSegmentationDataset` and does the conversion while the input is still a numpy array.

**Comment by lewfish** [comment]
> @jamesmcclain I made a PR to this branch ([jamesmcclain#2](https://github.com/jamesmcclain/raster-vision/pull/2)) that uses the same type conversion code as `SemanticSegmentationDataset` and does the conversion while the input is still a numpy array.

That looks more sound.

**Review by lewfish — APPROVED** [review]
